lua/neotest-python/init.lua

```diff
@@ -149,7 +150,8 @@ function PythonNeotestAdapter.build_spec(args)
     python_script,
     script_args,
   })
-  local strategy_config = get_strategy_config(args.strategy, python, python_script, script_args)
+  local env = { PYTHONPATH = base.get_python_path(position.path, vim.loop.cwd()) }
+  local strategy_config = get_strategy_config(args.strategy, python, python_script, script_args, env)
```

Comment:
We shouldn't rely on cwd. Search for pyproject.toml like it is done in lsp config.

Reply:
Reusing the lspconfig method is not possible:
```lua
local root = require("lspconfig.util").root_pattern("pyproject.toml")()
```
```
E5560: vimL function must not be called in a lua loop callback
```

Reply:
Using the following works
```lua
local function find_pyproject_root(fname)                                                                                                                                                                                                                           
  local is_pyproject = function(path)                                                                                                                                                                                                                               
      return Path:new(path, "pyproject.toml"):exists()                                                                                                                                                                                                              
  end                                                                                                                                                                                                                                                               
  return require("lspconfig.util").search_ancestors(fname, is_pyproject)                                                                                                                                                                                                                 
end
```